### ISSUE TITLE
Rustdoc does not properly escape colons in hrefs

### ISSUE THREAD

**Issue by Gankro** [rust-lang/rust#26842]
Given

```
# Example 2: aliasing a mutable reference
```

Rustdoc creates the following anchor

```
#example-2:-aliasing-a-mutable-reference
```

which browsers (Firefox at least) vomit on if you try to enter it the URL bar.

I suggest just eating the colon.

**Comment by smerrell** [comment]
This bug appears to have been fixed by b8cfa59be0a6b5200e525ff68390ede34698e909

**Comment by alexcrichton** [comment]
Yay! Thanks @smerrell!
